src/compiler/checker.ts

```diff
@@ -12315,6 +12319,10 @@ namespace ts {
             return getReducedType(getApparentType(getReducedType(type)));
         }
 
+        function isDiscriminableLiteralType(type: Type): boolean {
+            return isLiteralType(type) || isPatternLiteralType(type) || type === uniqueLiteralType || (!!(type.flags & TypeFlags.Intersection) && some((type as IntersectionType).types, isDiscriminableLiteralType));

```

Comment:
I would break the line here. It's too long to fit on any of my screens as-is.

```suggestion
            return isLiteralType(type) || isPatternLiteralType(type) || type === uniqueLiteralType 
                || (!!(type.flags & TypeFlags.Intersection) && some((type as IntersectionType).types, isDiscriminableLiteralType));
```